security.php

```diff
@@ -54,56 +57,202 @@ function wpcom_vip_is_restricted_username( $username ) {
 		|| WPCOM_VIP_MACHINE_USER_EMAIL === $username;
 }
 
+/**
+ * PRIVATE: This function is not intended to be used outside of this file and is subject to change without notice.
+ *
+ * Returns the defualts for limits.
+ */
+function _vip_get_auth_count_limit_defaults() {
+	if ( Context::is_fedramp() ) {
+		return [
+			'ip_username_login'          => 3,
+			'ip_login'                   => 5,
+			'username_login'             => 10,
+			'ip_username_password_reset' => 3,
+			'ip_password_reset'          => 3,
+			'username_password_reset'    => 15,
+
+			'ip_username_window'         => MINUTE_IN_SECONDS * 15,
+			'ip_window'                  => MINUTE_IN_SECONDS * 15,
+			'username_window'            => MINUTE_IN_SECONDS * 15,
+
+			'ip_username_lockout'        => MINUTE_IN_SECONDS * 30,
+			'ip_lockout'                 => MINUTE_IN_SECONDS * 30,
+			'username_lockout'           => MINUTE_IN_SECONDS * 30,
+		];
+	}
+
+
+	return [
+		'ip_username_login'          => 5,
+		'ip_login'                   => 50,
+		'username_login'             => 25,
+		'ip_username_password_reset' => 3,
+		'ip_password_reset'          => 3,
+		'username_password_reset'    => 15,
+
+		'ip_username_window'         => MINUTE_IN_SECONDS * 5,
+		'ip_window'                  => MINUTE_IN_SECONDS * 60,
+		'username_window'            => MINUTE_IN_SECONDS * 25,
+
+		'ip_username_lockout'        => MINUTE_IN_SECONDS * 5,
+		'ip_lockout'                 => MINUTE_IN_SECONDS * 60,
+		'username_lockout'           => MINUTE_IN_SECONDS * 25,
+	];
+}
+
+/**
+ * PRIVATE: This function is not intended to be used outside of this file and is subject to change without notice.
+ *
+ * Returns the cache keys used for login and forgotten passoword rate limiting.
+ *
+ * @param string $raw_username The username to use for the cache keys.
+ * @return array Associative array of cache keys.
+ */
+function _vip_login_cache_keys( $raw_username ) {
+	$username = vip_strict_sanitize_username( $raw_username );
+
+	// phpcs:ignore WordPressVIPMinimum.Variables.ServerVariables.UserControlledHeaders
+	$ip = filter_var( $_SERVER['REMOTE_ADDR'] ?? '', FILTER_VALIDATE_IP, [ 'options' => [ 'default' => '' ] ] );
+
+	return [
+		'ip_username_cache_key' => $ip . '|' . $username,
+		'ip_cache_key'          => $ip,
+		'username_cache_key'    => $username,
+	];
+}
+
+/**
+ * PRIVATE: This function is not intended to be used outside of this file and is subject to change without notice.
+ *
+ * Checks if counts for the given username and IP address are above the given thresholds.
+ * If so, the account is locked for a period of time.
+ *
+ * @param string $username The username to check.
+ * @param string $cache_group The cache group to use for the rate limiting.
```

Comment:
```suggestion
 * Check if counts for the given username and IP address are above the given thresholds.
 * If so, the account is locked for a period of time.
 *
 * @internal
 *
 * @param string $username The username to check.
 * @param string $cache_group The cache group to use for the rate limiting.
```

Why would this be considered private/internal?